cpp/src/parquet.cpp

```diff
@@ -427,21 +427,14 @@ ParquetReadInfo get_parquet_info(const std::string& glob_string,
   std::vector<std::string> file_paths = parse_glob(glob_string);
   if (file_paths.empty()) { throw std::invalid_argument("no parquet files specified"); }
 
-  // TODO: Using the default memory pool, hopefully it doesn't matter anyway here.
-  auto pool = arrow::default_memory_pool();
-
   // Open the first file to get schema information
   auto reader = ARROW_RESULT(arrow::io::ReadableFile::Open(file_paths[0]));
 
-  // Newer versions arrow have versions that return a result which is more convenient (also below)
-  std::unique_ptr<parquet::arrow::FileReader> parquet_reader;
-  auto status = parquet::arrow::OpenFile(reader, pool, &parquet_reader);
-  if (!status.ok()) {
-    throw std::runtime_error("failed to open parquet file: " + status.ToString());
-  }
+  // TODO: Using the default memory pool, hopefully it doesn't matter anyway here.
+  auto parquet_reader =
+    ARROW_RESULT(parquet::arrow::OpenFile(reader, arrow::default_memory_pool()));
   std::shared_ptr<arrow::Schema> schema;
-  status = parquet_reader->GetSchema(&schema);
-  if (!status.ok()) { throw std::runtime_error("failed to get schema: " + status.ToString()); }
+  auto status = parquet_reader->GetSchema(&schema);
```

Comment:
Is there a version now that allows:
```suggestion
  auto schema = ARROW_RESULT(parquet_reader->GetSchema());
```
otherwise, I think the status check still makes sense (if unlikely to fail).

Reply:
No they didn't update this function in arrow yet.